tools/pycbc_test_suite.sh

```diff
@@ -11,77 +11,63 @@ LOG_FILE=$(mktemp -t pycbc-test-log.XXXXXXXXXX)
     do
         echo -e ">> [`date`] running unit test for $prog"
         python $prog &> $LOG_FILE
-        if test $? -ne 0 ; then
-            RESULT=1
-            echo -e "    FAILED!"
-            echo -e "---------------------------------------------------------"
-            cat $LOG_FILE
-            echo -e "---------------------------------------------------------"
-        else
-            echo -e "    Pass."
-        fi
+	test_result
     done
 fi
 
 if [ "$PYCBC_TEST_TYPE" = "help" ] || [ -z ${PYCBC_TEST_TYPE+x} ]; then
     # check that all executables that do not require
     # special environments can return a help message
-    for prog in `find ${PATH//:/ } -maxdepth 1 -name 'pycbc*' -print 2>/dev/null | egrep -v '(pycbc_live_nagios_monitor|pycbc_make_offline_grb_workflow|pycbc_mvsc_get_features|pycbc_upload_xml_to_gracedb|pycbc_coinc_time)' | sort | uniq`
+    for prog in `find ${PATH//:/ } -maxdepth 1 -name 'pycbc*' -print 2>/dev/null | egrep -v '(pycbc_live_nagios_monitor|pycbc_mvsc_get_features|pycbc_coinc_time)' | sort | uniq`
     do
         echo -e ">> [`date`] running $prog --help"
         $prog --help &> $LOG_FILE
-        if test $? -ne 0 ; then
-            RESULT=1
-            echo -e "    FAILED!"
-            echo -e "---------------------------------------------------------"
-            cat $LOG_FILE
-            echo -e "---------------------------------------------------------"
-        else
-            echo -e "    Pass."
-        fi
+	test_result
+	if [[ `echo $prog | egrep '(pycbc_copy_output_map|pycbc_submit_dax|pycbc_stageout_failed_workflow)'` ]] ; then
+	    continue
+	fi
+	echo -e ">> [`date`] running $prog --version"
+	$prog --version &> $LOG_FILE
+	test_result
+    done
+    # also check that --version with increased modifiers works for one executable
+    echo -e ">> [`date`] running pycbc_inspiral --version with modifiers"
+    for modifier in "" 0 1 2 3
+    do
+	echo -e ">> [`date`] running pycbc_inspiral --version ${modifier}"
+        pycbc_inspiral --version ${modifier} &> $LOG_FILE
+        test_result
     done
-    # also check that --version works for one executable
-    echo -e ">> [`date`] running pycbc_inspiral --version"
-    pycbc_inspiral --version &> $LOG_FILE
-    if test $? -ne 0 ; then
-        RESULT=1
-        echo -e "    FAILED!"
-        echo -e "---------------------------------------------------------"
-        cat $LOG_FILE
-        echo -e "---------------------------------------------------------"
-    else
-        echo -e "    Pass."
-    fi
 fi
 
 if [ "$PYCBC_TEST_TYPE" = "search" ] || [ -z ${PYCBC_TEST_TYPE+x} ]; then
     # run pycbc inspiral test
     pushd examples/inspiral
     bash -e run.sh
-    if test $? -ne 0 ; then
-        RESULT=1
-        echo -e "    FAILED!"
-        echo -e "---------------------------------------------------------"
-    else
-        echo -e "    Pass."
-    fi
+    test_result
```

Comment:
`test_result` will try to cat `$LOG_FILE` which does not exist for this test. Maybe you could add a check in `test_result` that only cats it if it exists? That may still cat it in unwanted situations though, so maybe an argument is better?

🐈

Reply:
I added a :black_cat:`_output` global variable which gets checked whether to give log file output, and this is set to false when we aren't checking the output any more.

If we ever want to have a mix though, we need to remember to set/unset it as appropriate